src/js/containers/recipientLanding/RecipientLandingContainer.jsx

```diff
@@ -0,0 +1,166 @@
+/**
+ * RecipientLandingContainer.jsx
+ * Created by David Trinh 7/2/18
+ */
+
+import React from 'react';
+import { isCancel } from 'axios';
+import { inRange } from 'lodash';
+
+
+import * as RecipientLandingHelper from 'helpers/recipientLandingHelper';
+
+import RecipientLandingContent from 'components/recipientLanding/RecipientLandingContent';
+
+import BaseRecipientLandingRow from 'models/v2/recipient/BaseRecipientLandingRow';
+
+
+export default class RecipientLandingContainer extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            pageNumber: 1,
+            order: {
+                field: 'name',
+                direction: 'asc'
+            },
+            inFlight: false,
+            error: false,
+            searchString: '',
+            results: [],
+            totalItems: 0,
+            pageSize: 50
+        };
+
+        this.recipientsRequest = null;
+        this.setRecipientSearchString = this.setRecipientSearchString.bind(this);
+        this.onChangePage = this.onChangePage.bind(this);
+        this.setSort = this.setSort.bind(this);
+    }
+
+    componentDidMount() {
+        this.fetchRecipients();
+    }
+
+    componentWillUnmount() {
+        if (this.recipientsRequest) {
+            this.recipientsRequest.cancel();
+        }
+    }
+
+    onChangePage(pageNumber) {
+        const totalPages = Math.ceil(this.state.totalItems / this.state.pageSize);
+        if (inRange(pageNumber, 1, totalPages + 1)) {
+            // Change page number in the state and make a new request
+            this.setState({
+                pageNumber
+            }, () => {
+                this.fetchRecipients();
+            });
+        }
+    }
+
+    setRecipientSearchString(searchString) {
+        // Change search string in the state and make a new request
+        if (searchString.length > 2) {
+            this.setState({
+                searchString
+            }, () => {
+                this.fetchRecipients();
+            });
+        }
+    }
+
+    setSort(field, direction) {
+        // Change sort in the state and make a new request
+        // Reset the page number to 1
+        this.setState({
+            order: {
+                field,
+                direction
+            },
+            pageNumber: 1
+        }, () => {
+            this.fetchRecipients();
+        });
+    }
+
+    fetchRecipients() {
+        if (this.recipientsRequest) {
+            // a request is in-flight, cancel it
+            this.recipientsRequest.cancel();
+        }
+
+        this.setState({
+            inFlight: true,
+            error: false
+        });
+
+        // generate the params
+        const pageSize = 50;
+        const params = {
+            order: this.state.order.direction,
+            sort: this.state.order.field,
+            page: this.state.pageNumber,
+            limit: pageSize
```

Comment:
We can use `this.state.pageSize` here so that if the default page size ever changes, we'll only have to update it in one place.